robot/session_manager.go

```diff
@@ -104,6 +106,16 @@ func (m *SessionManager) expireLoop(ctx context.Context) {
 					}()
 					res, err := m.robot.ResourceByName(resName)
 					if err != nil {
+						// It's possible at this point that the robot is Closing, the
+						// resource manager has already been closed, and the resource
+						// associated with the session has been removed from the graph and
+						// cannot be found. If the error is a not found error and the
+						// context has errored, return without appending to resourceErrs
+						// and set serverClosing to true.
+						if strings.Contains(err.Error(), resource.NewNotFoundError(resName).Error()) && ctx.Err() != nil {
```

Comment:
Instead of doing string checking which can fail more easily, would you be open to making a `IsNotAvailableError` like error for `IsNotFoundError`?

Reply:
Alternatively, just check if ctx Error is not nil, return here, check ctx Error != nil again, return, etc..

Reply:
> would you be open to making a IsNotAvailableError like error for IsNotFoundError

yep! good idea.

>  just check if ctx Error is not nil

I'd rather check that the error is specifically resource not found and the context has errored. `ResourceByName` can also return a `NotAvailableError` or a `RemoteResourceClashError`, and it's not 100% clear to me that we want to suppress those because the robot is closing.

Reply:
sgtm